src/cross_section.F90

```diff
@@ -526,6 +572,185 @@ pure function find_energy_index(mat, E) result(i)
+    real(8), intent(in)              :: Emev                         !< The energy at which to evaluate the cross section in MeV
+    real(8), intent(in)              :: sqrtkT                       !< The temperature in the form sqrt(kT (in eV)), at which to evaluate the cross section.
+    real(8), intent(out)             :: sigT                         !< Total cross section
+    real(8), intent(out)             :: sigA                         !< Absorption cross section
+    real(8), intent(out)             :: sigF                         !< Fission cross section
+    complex(8) :: PSIIKI
+    complex(8) :: CDUM1
+    complex(8) :: w_val
+    complex(8) :: Z
+    real(8) :: sqrtE
+    real(8) :: invE
+    real(8) :: DOPP
+    real(8) :: DOPP_ECOEF
+    real(8) :: temp
+    real(8) :: E
+    integer :: iP
+    integer :: iC
+    integer :: iW
+    integer :: startw
+    integer :: startw_1
+    integer :: startw_endw
+    integer :: endw
+
+    ! Convert to eV
+    E = Emev * 1.0D6
+
+    sqrtE = sqrt(E)
+    invE = E**(-1)
+
+    if(mp_already_alloc .eqv. .FALSE.) then
+      call multipole_eval_allocate()
+    end if
+
+    ! Locate us
+    iW = floor((sqrtE - sqrt(multipole % start_E))/multipole % spacing + 1.0_8)
+
+    startw = multipole % w_start(iW)
+    startw_1 = startw - 1 ! This is an index shift parameter.
+    endw = multipole % w_end(iW)
+    startw_endw = endw - startw + 1
+
+    ! Fill in factors
+    if (startw <= endw) then
+      call fill_factors(multipole, sqrtE, sigT_factor, twophi, multipole % num_l)
+    end if
+
+    ! Generate some doppler broadening parameters
+
+    ! DOPP_ECOEF is inverse of dopp, divided by E, multiplied by sqrt(pi).
+    DOPP = multipole % sqrtAWR/sqrtKT
+    DOPP_ECOEF = DOPP*invE*SQRT_PI
+
+    sigT = 0.0_8
+    sigA = 0.0_8
+    sigF = 0.0_8
+    ! Evaluate linefit first
+
+    if(sqrtkT /= 0 .AND. multipole % broaden_poly(iW) == 1) then ! Broaden the curvefit.
+      call broaden_n_polynomials(E, DOPP, multipole % fit_order + 1, broadened_polynomials)
+
+      do iC = 1,multipole % fit_order+1
+        sigT = sigT + multipole % curvefit(FIT_T, iC, iW)*broadened_polynomials(iC)
+        sigA = sigA + multipole % curvefit(FIT_A, iC, iW)*broadened_polynomials(iC)
+        if (multipole % fissionable) then
+          sigF = sigF + multipole % curvefit(FIT_F, iC, iW)*broadened_polynomials(iC)
+        end if
+      end do
+    else ! Evaluate as if it were a polynomial
+      temp = invE
+      do iC = 1,multipole % fit_order+1
+
+        sigT = sigT + multipole % curvefit(FIT_T, iC, iW)*temp
+        sigA = sigA + multipole % curvefit(FIT_A, iC, iW)*temp
+        if (multipole % fissionable) then
+          sigF = sigF + multipole % curvefit(FIT_F, iC, iW)*temp
+        end if
+
+        temp = temp * sqrtE
+      end do
+    end if
+
+    ! Then get the poles we want and broaden them.
+
+    if (sqrtkT == 0.0_8) then
```

Comment:
The Faddeeva function used, as far as I am aware, will give results accurate to 14 digits across the entire domain of interest.  The only reason an asymptotic form is necessary is that the z in w(z) will be infinity at 0K.  I remember testing non-asymptotic accuracy down to 0.1 K and having no issues.